homeassistant/components/zamg/config_flow.py

```diff
@@ -72,14 +72,13 @@ async def async_step_user(
             )
 
         return self.async_create_entry(
-            title=user_input.get(CONF_NAME) or self._client.get_station_name,
+            title=self._client.get_station_name,
             data={CONF_STATION_ID: station_id},
         )
 
     async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
         """Handle ZAMG configuration import."""
         station_id = str(config.get(CONF_STATION_ID))
```

Comment:
```suggestion
        station_id = str(config[CONF_STATION_ID])
```

Reply:
Why are we swapping back and forth between integers and strings for the station id?

Reply:
The API needs it as string

Reply:
If the API returns a string and we store a string in the config entry data, why do we convert it to an integer in the user form?